src/BaseBinaryStar.cpp

```diff
@@ -1390,366 +1390,388 @@ void BaseBinaryStar::ResolveCoalescence() {
+            m_PsiE  = _2_PI * RAND->Random(); 
+        }
+        else {                     
 
-    m_Supernova->SetPostSNeOrbitalEnergy(CalculateOrbitalEnergy(reducedMassPrime, totalMassPrime, m_SemiMajorAxisPrime));           // post-SN orbital energy, check if still bound
-    double epsilon     = -m_Supernova->PostSNeOrbitalEnergy() / m_Supernova->PreSNeOrbitalEnergy();                                 // dimensionless post-SN orbital energy
+            ////////////////////////////////////////
+            // 
+            // Binary is still bound 
+            // 
+            ////////////////////////////////////////
 
-    m_CosIPrime        = 0.0;
-    m_IPrime           = 0.0;
-    m_SystemicVelocity = 0.0;
+            // Set the component velocites to the system velocity (to simplify the UpdateComponentVelocity function). System velocity was already correctly set above.
+             
+            m_Supernova->UpdateComponentVelocity( Vcm_.RotateVector(m_ThetaE, m_PhiE, m_PsiE) );
+            m_Companion->UpdateComponentVelocity( Vcm_.RotateVector(m_ThetaE, m_PhiE, m_PsiE) );
 
-    if (utils::Compare(epsilon, 0.0) < 0) {		                                                                                    // still bound?
+            ////////////////////////////////////////////////////////////////////////////////////
+            // Calculate Euler angles - see RotateVector() in vector.cpp for details
 
-        // Calculate post-SN orbital inclination using the equation for arbitrary eccentricity orbits
-        m_CosIPrime   = CalculateCosFinalPlaneTilt(m_Supernova->SN_Theta(), m_Supernova->SN_Phi());
-        m_IPrime      = acos(m_CosIPrime);
+            m_ThetaE = angleBetween( H/h, H_/h_ );                         // Angle between the angular momentum unit vectors, always well defined
 
-        m_SystemicVelocity = CalculatePostSNSystemicVelocity(m_Supernova->Mass(),                                                   // post-SN systemic (center-of-mass) velocity in ms s^-1
-                                                             m_Supernova->MassPrev() - m_Supernova->Mass(),
-                                                             m_Companion->Mass(),
-                                                             totalMass,
-                                                             totalMassPrime,
-                                                             m_Supernova->SN_Theta(),
-                                                             m_Supernova->SN_Phi());
-        m_SystemicVelocity /= KM;                                                                                                   // convert to km s^-1
-    }
-    else {                                                                                                                          // no longer bound
-        m_Unbound = true;
-    }
+            // If N = H x H_ is not well-defined, need to account for degeneracy between eccentricities
```

Comment:
If H is parallel or anti-parallel to H_ (H prime), then the cross product is not well-defined (e.g if the binary orbital plane is the same, or completely inverted, during the SN).

The degeneracy between eccentricities refers to the special case where the cross product isn't well-defined AND both ecc<SN and ecc>SN are >0. Then the individual Euler angles phi and psi are not defined (since they are defined relative to the N vector), but either the sum or difference is (for parallel and anti-parallel respectively). 

I don't know any literature that refers to this, it's not really an open problem. It might be a good idea to put this into the COMPAS methods paper, so that it's clear (to us all of us and everyone else) what's happening.